### DISCUSSION TITLE
Interested in Adding Extensive Theming Options to RefindPlus?

### DISCUSSION THREAD

**Discussion by raldone01** [dakanji/RefindPlus#103] [General]
I created this in response to this [upstream merge request](https://sourceforge.net/p/refind/code/merge-requests/41/).
I would love to add more extensive config options for theming. As I am more familiar with github and I think this would fit the spirit of RefindPlus - would you be interested in a theming focused pull request in the future? (You can also implement these by yourself too.)

I would like to discuss how these options should be named and what should be adjustable.

**Comment by dakanji** [comment]
Hi, I didn't quite fully understand how you envisage this when I saw it upstream earlier. What is it meant to achieve and how do you see it being done?

Have to let you know that I prefer that big changes be proposed upstream as I cling to a probably out of step idea of largely staying in sync with the upstream project. Anyway, let's see what you have in mind.

**Reply by raldone01** [reply]
I see. In that case we can continue the conversation upstream. I thought it may be better to do it here if you were going to implement it in RefindPlus anyways to avoid two different implementations of set of features.

**Comment by dakanji** [comment]
My reference for implementing upstream on big stuff is to just to reduce the stuff going in here but not there.

So it is fine with me if you prefer to go upstream. Only issue with upstream is the lack of activity and not knowing if/when that will change.

**Comment by raldone01** [comment]
Actually I would prefer to get my stuff merged into your fork where it would be more likely to be actually used. If our conversation on upstream turns into something you would want in your fork I might directly base my work on your fork otherwise I will code it based on upstream.

You should also consider to modify the docs with your options in addition to the readme and maybe host the modified docs website on the github user pages.

**Comment by dakanji** [comment]
> Actually I would prefer to get my stuff merged into your fork

Gut feel is that given the nature of the changes, the proposal would have to be adopted upstream first.

> You should also consider to modify the docs with your options in addition to the readme and maybe host the modified docs website on the github user pages.

The main thing I feel needs being done is syncing the installation stuff to work on RP. This is because I think users can get the documentation info from upstream for now and the options are explained in the config file. Hosting docs is something to consider regardless though. I suppose both are things I should put somewhere as `Help-Wanted` items.

**Comment by raldone01** [comment]
> Gut feel is that given the nature of the changes, the proposal would have to be adopted upstream first.

I understand. Thanks for your input. At this point it may be best to go ahead with the addition of the two options you mentioned.
I will see how many changes are a needed to make my proposal work.

**Comment by dakanji** [comment]
Good luck. Can always take another look down the line.
